src/main/resources/liquibase/tenant/scripts/2020-07-16--10-00-create-instnace-ids-table.xml

```diff
@@ -18,10 +18,50 @@
     </createTable>
   </changeSet>
 
+  <changeSet id="2020-11-19--18-00-change-json-to-text-column-in-instances-table" author="Illia Daliek">
+    <modifyDataType tableName="instances" columnName="json" newDataType="text" schemaName="${database.defaultSchemaName}"/>
+  </changeSet>
+
+  <changeSet id="2020-11-19--18-00-migrate-json-to-text" author="Illia Daliek">
+    <sql>ALTER TABLE ${database.defaultSchemaName}.instances ALTER COLUMN json TYPE text</sql>
+  </changeSet>
+
+  <changeSet id="2020-11-19--18-00-change-varchar-to-uuid-column-in-instances-table" author="Illia Daliek">
+    <modifyDataType tableName="instances" columnName="request_id" newDataType="uuid"/>
+  </changeSet>
+
+  <changeSet id="2021-01-19--14-00-add-index-to-instances-table-on-instance_id-column" author="Illia Daliek">
+    <preConditions onFail="MARK_RAN">
+      <not>
+        <indexExists tableName="instances" indexName="instance_id_idx"/>
+      </not>
+    </preConditions>
+    <createIndex tableName="instances" indexName="instance_id_idx" unique="false" >
+      <column name="instance_id"/>
+    </createIndex>
+  </changeSet>
+
+  <changeSet id="2021-02-23--13-00-add-index-to-instances-table-on-request_id-column" author="Illia Daliek">
+    <preConditions onFail="MARK_RAN">
+      <not>
+        <indexExists tableName="instances" indexName="request_id_idx"/>
+      </not>
+    </preConditions>
+    <createIndex tableName="instances" indexName="request_id_idx" unique="false">
+      <column name="request_id"/>
+    </createIndex>
+  </changeSet>
+
   <changeSet id="2021-02-24--12-00-add-autoIncrement-pk" author="Illia Daliek">
     <addColumn tableName="instances">
       <column name="id" type="integer" autoIncrement="true"/>
     </addColumn>
   </changeSet>
 
+  <changeSet id="2021-03-29--14-00-add-index-on-id-pk-field" author="Illia Daliek">
+    <createIndex tableName="instances" indexName="instance_id_pk_idx">
```

Comment:
why not add the precondition here too?

Reply:
@craigmcnally, I moved some scripts from files and they were already applied at environments, but the scripts moving triggered their execution again which leads to an error. The instance_id_pk_idx is totally new here and has not been executed anywhere, that is why it is not required the precondition.

Reply:
not required, but it can't hurt, In the case where maybe the module is enabled, then disabled, and later enabled again I think not having this here would be problematic.

Reply:
Okay, I added the precondition.